packages/paste-website/src/pages/components/chat-composer/index.mdx

```diff
@@ -163,14 +167,94 @@ plugin `AutoScrollPlugin`.
     ChatAttachmentLink,
     ChatMessageMetaItem,
     ChatComposer,
-    AutoScrollPlugin,
     Button,
     SendIcon,
+    SendButtonPlugin,
+    EnterKeySubmitPlugin,
+    createNewMessage,
+    $getRoot,
+    useChatLogger,
+    DownloadIcon,
+    ClearEditorPlugin,
   }}
 >
   {ChatDialogExample}
 </LivePreview>
 
+### Adding interactivity with Plugins
+
+In the above example we make use of three Lexical Plugins, `ClearEditorPlugin` that is provided by Lexical, and two custom plugins `SendButtonPlugin` and `EnterKeySubmitPlugin`. We also keep track of the content provided to the composer via the `onChange` handler. Together we can add custom interactivity such as:
+
+- Clear the editor on button click using `ClearEditorPlugin`
+- Submit on enter key press using `EnterKeySubmitPlugin`
+- Submit on button click using `SendButtonPlugin`
+
+Plugins are functions that must be children of the `ChatComposer` component, so that they can access the Composer context.
+
+#### `onChange` event handler
+
+The `onChange` handler provided to the `ChatComposer` takes three arguments, the first of which is the [`editorState`](https://lexical.dev/docs/concepts/editor-state#understanding-the-editor-state). This allows us to read the current content of the editor using the utilities provided by Lexical.
+
+`$getRoot` is a utility to access the composer root [`ElementNode`](https://lexical.dev/docs/api/classes/lexical.ElementNode). We can then get the text content of the editor everytime it is updated, and store it in our component state for later use.
+
+```js
+const handleComposerChange = (editorState: EditorState): void => {
+  editorState.read(() => {
+    const text = $getRoot().getTextContent();
+    setMessage(text);
+  });
+};
+```
+
+#### `ClearEditorPlugin`
+
+The `ClearEditorPlugin` supplied by Lexical allows you to build functionality into the composer that will clear the composer content when a certain action is performed.
+
+When passed as a child to `ChatComposer` it will automatically register a `CLEAR_EDITOR_COMMAND`. You can then dispatch this command from elsewhere to clear the composer content. As example, we created two plugins; `SendButtonPlugin` and `EnterKeySubmitPlugin` which both dispatch the `CLEAR_EDITOR_COMMAND`, and clear the composer content as a result.
```

Comment:
```suggestion
When passed as a child to `ChatComposer`, it will automatically register a `CLEAR_EDITOR_COMMAND`. You can then dispatch this command from elsewhere to clear the composer content. In the example, we created 2 plugins: `SendButtonPlugin` and `EnterKeySubmitPlugin` which both dispatch the `CLEAR_EDITOR_COMMAND`, and clear the composer content as a result.
```